### PR TITLE
Refactor and unify Field / FastField test suite

### PR DESCRIPTION
**Pull request by prichodko**
It feels like cheating.

### CODE REVIEW

File: test/Field.test.tsx [hunk 1]
```diff
@@ -1,545 +1,352 @@
...
-import { shallow, mount } from 'enzyme';
 import { noop } from './testHelpers';
 
-interface TestFormValues {
-  name: string;
-  email: string;
+const initialValues = { name: 'jared', email: 'hello@reason.nyc' };
+type Values = typeof initialValues;
+
+function renderForm(ui: any, props?: Partial<FormikConfig<Values>>) {
```
Comment (jaredpalmer) [inline test/Field.test.tsx]: This needs to be typed.

Comment (jaredpalmer) [inline test/Field.test.tsx]:
```suggestion
function renderForm(ui: any, props?: Partial<FormikConfig<Values>>) {
function renderForm(ui: React.ReactNode, props?: Partial<FormikConfig<Values>>) {
```

Comment (jaredpalmer) [reply test/Field.test.tsx]: I also think we should maybe make both optional and reverse the order

Comment (jaredpalmer) [reply test/Field.test.tsx]: We should also share these between Field and Formik in a perfect world

File: test/Field.test.tsx [hunk 2]
```diff
@@ -1,545 +1,352 @@
...
+
+function cases(
+  title: string,
+  tester: (arg: typeof renderField | typeof renderFastField) => void
+) {
+  describe(title, () => {
+    it('<Field />', async () => Promise.resolve(tester(renderField)));
+    it('<FastField />', async () => Promise.resolve(tester(renderFastField)));
+  });
+}
```
Comment (jaredpalmer) [inline test/Field.test.tsx]: Cool

Comment (jaredpalmer) [reply test/Field.test.tsx]: TIL!

File: test/Field.test.tsx (hunk 1 above)
Comment (prichodko) [reply test/Field.test.tsx]: I wanted to keep it as close as possible to original `render` from `react-testing-library`, e.g. ui as a first argument.

File: test/Field.test.tsx (hunk 2 above)
Comment (prichodko) [reply test/Field.test.tsx]: Yeah, I was thinking if we could use `Promise.resolve` trick in Formik instead of `isPromise` check.

**Comment by prichodko** [comment]
I will definitely look into further optimization of renderers when it all gets refactored / merged. 

Regarding the order, we will probably see what is used more, if passing props or rendering ui. Because if it is `ui`, then it would be cumbersome to always pass `{}` as a first arg.

File: test/Field.test.tsx [hunk 3]
```diff
@@ -1,545 +1,357 @@
...
-      const FormFields = () => <Field name="name" validate={validate} />;
-
-      const tree = mount(<TestForm component={FormFields} />);
-      tree
-        .find(FormFields)
-        .props()
-        .validateField('name');
-      await Promise.resolve()
-        .then()
-        .then(); // XXX: Waiting for validateField's promise, should be fixed :(
```
Comment (prichodko) [inline test/Field.test.tsx]: Fixed 😆

**Pull request merged**